### ISSUE TITLE
[linuxd] Separate Test Cases in `linux-app` into Multiple Files

### ISSUE THREAD

**Issue by ppenna** [ppenna/nanvix#329]
## Context

The `linux-app` is becoming cluttered: I am adding every new test case in `main()`.

## Proposed Solution

We should separate the test cases from `main()`. Ideally, we want to have one submodule for each subsystem, with one testing function for each test case.

## Code References

https://github.com/nanvix/nanvix/blob/30482b8359d9bcb3eef86a107084490556c43b95/src/benchmarks/linux-app/src/main.rs#L35

**Comment by LucasGdosR** [comment]
Splitting this function is tricky, because many tests need to run serially (like create a file → write → seek → close; create → rename; create → link → readlink →unlink), and splitting into functions makes them run in parallel, from my understanding. There's also the issue of logging in parallel leading to an incomprehensible log in arbitrary order.

1. Should I enforce an ordering on the tests / run them serially?
2. Should I include a kind of setup/teardown for the tests that rely on temporary files and create a file for each? Is there a particular way I should use, or can I just make it my way?
3. Should I run them in parallel? I could create a locked counter to generate unique `VirtualEnvironmentIdentifier`s and use them to name the created temporary files and directories, so side effects from one function don't interact with another.
4. Is there another detail I'm missing?

**Comment by ppenna** [comment]
> Splitting this function is tricky, because many tests need to run serially (like create a file → write → seek → close; create → rename; create → link → readlink →unlink), and splitting into functions makes them run in parallel, from my understanding. 

Yes, they need to run serially. You should look for encapsulating different tests in composable ones. For instance, you can have a test to `create` and `unlink`,and compose it with `read` /`write`.

> There's also the issue of logging in parallel leading to an incomprehensible log in arbitrary order.
You don't need to worry about parallel execution of tests here.

> Sould I enforce an ordering on the tests / run them serially?

Yes, you should, when testing a particular system call. For instance, if you are testing whether you can `unlink` a file that exist, you should have called `create` before.

> Should I include a kind of setup/teardown for the tests that rely on temporary files and create a file for each? Is there a particular way I should use, or can I just make it my way?

Each test must ideally operate on a fresh state. How you do that is up to you.

> Should I run them in parallel? I could create a locked counter to generate unique VirtualEnvironmentIdentifiers and use them to name the created temporary files and directories, so side effects from one function don't interact with another.

This issue is not about that. So, you should not.

> Is there another detail I'm missing?

You are probably going more complex than it is required. This issue is about spliting the tests that are in the same function into multiple functions, potentially in multiple files. Preserve the serial behavior and just copy and paste as much as possible.
